uaclient/cli.py

```diff
@@ -1287,6 +1287,9 @@ def action_attach(args, *, cfg):
         event.process_events()
         return 1
     else:
+        current_iid = identity.get_instance_id()
+        if current_iid:
+            cfg.write_cache("instance-id", current_iid)
```

Comment:
Instead of adding this here. Lets do something like this in `actions.py`
```diff
diff --git a/uaclient/actions.py b/uaclient/actions.py
index 9af063a6..9b950762 100644
--- a/uaclient/actions.py
+++ b/uaclient/actions.py
@@ -44,6 +44,10 @@ def attach_with_token(
         update_apt_and_motd_messages(cfg)
         raise exc
 
+    current_iid = identity.get_instance_id()
+    if current_iid:
+        cfg.write_cache("instance-id", current_iid)
+
     update_apt_and_motd_messages(cfg)
 
 
@@ -69,9 +73,6 @@ def auto_attach(
                 ua_status.MESSAGE_UNSUPPORTED_AUTO_ATTACH
             )
         raise e
-    current_iid = identity.get_instance_id()
-    if current_iid:
-        cfg.write_cache("instance-id", current_iid)
 
     token = tokenResponse["contractToken"]
 

```

Reply:
Considering that auto-attach does that in the `action.auto_attach` method, I think this will mean we run the same code snippet twice when running the auto-attach command. Unless, we also remove the same snippet from the auto_attach function and move it into the `attach_with_token` function.

Reply:
@orndorffgrant Please check the update.

Reply:
@CalvoM Change looks good! Now we just need to check for this write_cache call in test_actions.py:TestAttachWithToken instead of where it is currently failing in test_actions.py:TestAutoAttach